csFastFloat/Structures/DecimalInfo.cs

```diff
@@ -403,5 +389,120 @@ unsafe internal static DecimalInfo parse_decimal(char* p, char* pend, char decim
 
       return answer;
     }
+
+    // UTF-8/ASCII inputs
+    unsafe internal static DecimalInfo parse_decimal(byte* p, byte* pend, byte decimal_separator)
+    {
+      DecimalInfo answer = new DecimalInfo() { negative = (*p == '-'), digits = new byte[Constants.max_digits] };
+
+      if ((*p == '-') || (*p == '+'))
+      {
+        ++p;
+      }
+      // skip leading zeroes
+      while ((p != pend) && (*p == '0'))
+      {
+        ++p;
+      }
+      while ((p != pend) && Utils.is_integer(*p))
+      {
+        if (answer.num_digits < Constants.max_digits)
+        {
+          answer.digits[answer.num_digits] = (byte)(*p - '0');
+        }
+        answer.num_digits++;
+        ++p;
+      }
+      if ((p != pend) && (*p == decimal_separator))
+      {
+        ++p;
+        byte* first_after_period = p;
+        // if we have not yet encountered a zero, we have to skip it as well
+        if (answer.num_digits == 0)
+        {
+          // skip zeros
+          while ((p != pend) && (*p == '0'))
+          {
+            ++p;
+          }
+        }
+        while ((p + 8 <= pend) && (answer.num_digits + 8 < Constants.max_digits)) 
+        {
+          ulong val;
+          Buffer.MemoryCopy(p, &val, sizeof(ulong), sizeof(ulong));
```

Comment:
```suggestion
          ulong val = Unsafe<ulong>.Read(&p);
```
Not sure exactly what the intent of doing a memcopy for a ulong is though I'll admit.

Reply:
Switched to `Unsafe.ReadUnaligned<ulong>(chars)`.

Reply:
The idea here is to process 8 characters at once instead of doing byte-by-byte processing.

Reply:
I am resolving this conversation, please see updated code.